### PR TITLE
Finetune matrix transform kernel & related APIs

### PR DESCRIPTION
**Pull request by Serge45**
## Brief ##
Finetune `hipblasLtMatrixTransform` API and related documents

## Tasks ##
 - [x] Added empty `A` or empty `B` supports
 - [x] Added error handling for invalid configurations
 - [x] Added on device scalar support
 - [x] Related test cases for above items
 - [x] Integrated documents to document generating system

### CODE REVIEW

**Review by KKyang — APPROVED** [review]

**Review by KKyang — APPROVED** [review]

File: docs/functions.rst [hunk 1]
```diff
@@ -82,3 +82,35 @@ HIPBLASLT_COMPUTE_F32   HIPBLASLT_R_32F       HIPBLASLT_R_16F HIPBLASLT_R_16F
 HIPBLASLT_COMPUTE_F32   HIPBLASLT_R_32F       HIPBLASLT_R_16F HIPBLASLT_R_32F
 HIPBLASLT_COMPUTE_F32   HIPBLASLT_R_32F       HIPBLASLT_R_16B HIPBLASLT_R_16B
 ======================= =================== ============= ==============
+
```
Comment (jichangjichang) [inline docs/functions.rst:85]: You miss adding hipblasLtMatrixTransformDesc_t in documentation

Comment (jichangjichang) [reply docs/functions.rst:85]: @Serge45 what I means is that you should add hipblasLtMatrixTransformDesc_t into hipBLASLt Datatypes section in documentation.

Comment (Serge45) [reply docs/functions.rst:85]: Thanks for the reminder! amended.

**Review by jichangjichang — APPROVED** [review]
lgtm
please merge once CI get passed

File: library/include/hipblaslt.h [hunk 2]
```diff
@@ -883,19 +892,21 @@ hipblasStatus_t hipblasLtMatrixTransformDescSetAttribute( //
...
  *
- * \retval     HIPBLAS_STATUS_INVALID_VALUE  if sizeInBytes is 0 and sizeWritten is NULL, or if  sizeInBytes is non-zero
+ * \retval HIPBLAS_STATUS_INVALID_VALUE  if sizeInBytes is 0 and sizeWritten is NULL, or if  sizeInBytes is non-zero
  *                                          and buf is NULL or sizeInBytes doesn't match size of internal storage for
  *                                          selected attribute
- * \retval     HIPBLAS_STATUS_SUCCESS        if attribute's value was successfully written to user memory
+ * \retval HIPBLAS_STATUS_SUCCESS        if attribute's value was successfully written to user memory
  */
 HIPBLASLT_EXPORT
 hipblasStatus_t hipblasLtMatrixTransformDescGetAttribute( //
```
Comment (jichangjichang) [inline library/include/hipblaslt.h]: remove //

Comment (Serge45) [reply library/include/hipblaslt.h]: Amended.

**Pull request merged**